### DISCUSSION TITLE
Wireless works better than wired for some reason

### DISCUSSION THREAD

**Discussion by ghost** [nefarius/DsHidMini#136] [Support]
I just got this working after removing the official sony drivers and it all seems to work great (XInput mode not emulating an xbox controller aside) but in wired, every option just seems to read as "wireless controller for playstation 3"
![image](https://user-images.githubusercontent.com/39570503/131584352-c0a2c86e-dd82-4ef4-9475-1e4e5e6fd2b1.png)

Any ideas?

**Comment by nefarius** [comment]
So what? Does it work? The name is a residue from the Sony drivers and doesn't influence anything.

**Reply by ghost** [reply]
It works in the sense that it reads a controller is there pressing buttons but every mode does the exact same thing.

**Reply by Kanuan** [reply]
the name is mostly irrelevant.

A quick way to check what the controller really is doing is by testing it on [the gamepad tester website](https://gamepad-tester.com/). Different modes will appear with different layouts.

The exception is DS4Windows mode since it was designed to not be seen as a game controller. From an average end-user perspective it only is detected by Ryochan7's DS4Windows.

**Reply by ghost** [reply]
Thanks for the website rec, it's really cool! I had to change the mode around a few times to finally get it but the website does finally report my DS3 as XInput. Windows still calls it a "DS3 Compatible HID Device". This was all wired (before wired didn't behave at all so no idea what's happening) so I unplugged it and tried to see what would happen. 

Wirelessly, didn't matter what mode or how many times I flipped through them it was showing up as a PS3 controller (sometimes with mismapped buttons) until one time where I hit apply and it shut my controller off, turned it back on and now it's xinput perfectly. So I think issue-wise there's something funky going on with switching modes not happening reliably and the names reported to windows may confuse people like it did me.

**Reply by nefarius** [reply]
The device names are irrelevant, as mentioned. What matters is if the correct driver in the correct version has taken authority over the USB device. That's pretty much it. Any other weirdness is mostly due to some registry residue I have no control of, there are millions of installations of Windows out there, some more funky than others. And the modes behaving differently is literally what they are for 🙂 [We provide thorough documentation for that](https://vigem.org/projects/DsHidMini/HID-Device-Modes-Explained/).

In essence, it works now, did I get that right?

**Reply by Kanuan** [reply]
> Wirelessly, didn't matter what mode or how many times I flipped through them it was showing up as a PS3 controller (sometimes with mismapped buttons) until one time where I hit apply and it shut my controller off, turned it back on and now it's xinput perfectly

The "apply settings" button attempts to reconnect the controller (on bluetooth it just turns off). Sometimes it doesn't work. After changing the current HID device mode, manually reconnecting the controller should 100% work.

**Reply by ghost** [reply]
Makes sense. Another thing I tried was using "xinput" in a game that detects 360 controllers Rayman Origins. (XInput meaning the website says XInput and my PC calls it DS3 Compatible HID Device) and the game doesn't pick it up as XInput (XInput refers to button letters and automaps controls where DInput needs to be manually set and refers to numbers). So I don't know, is that expected or am I doing something wrong?

@nefarius I know they don't mean anything to those of us in the know, I just think it could cause confusion going forward. It doesn't seem to work as XInput but the other modes seem to work just fine.

**Reply by nefarius** [reply]
People are already confused and that's the fault of the mess that is the input API eco-system and every game engine handling things differently, except the way they should according to standards. There is nothing I can or will do about that, XI mode is in BETA for a reason, it does work, just not the way you might expect to which is, again, out of my reach. If you use this mode with Steam and force the game to use Steam Input, so far I got every game to work.

And DsHidMini is and was primarily for **emulators**, everything else is just a nice-to-have.

Cheers

**Reply by Kanuan** [reply]
The issues you are having are having are currently known. Check #114 

Unfortunately, for now if you want to use your DS3 with DsHidMini as a Xbox controller that works as "expected" from the average user perspective then you need to do Xbox Controller emulation through DS4Windows (read the [DS4Windows Mode User Guide](https://vigem.org/projects/DsHidMini/DS4-Mode-User-Guide/))

**Reply by nefarius** [reply]
> The issues you are having are having are currently known. Check #114

One "fix" I can think of is to spoof Microsoft Xbox One Controller VID and PID but I will not do that, these practices is the sole reason we have such a mess in the first place 😉

**Reply by ghost** [reply]
> 
> 
> The issues you are having are having are currently known. Check #114
> 
> Unfortunately, for now if you want to use your DS3 with DsHidMini as a Xbox controller that works as "expected" from the average user perspective then you need to do Xbox Controller emulation through DS4Windows (read the [DS4Windows Mode User Guide](https://vigem.org/projects/DsHidMini/DS4-Mode-User-Guide/))

That's fine, I'm not that hung up on it that I'd use 2 programs in tandem to do it. I'll keep it in mind though. I really reported because it said beta and I wanted to be helpful :) Thanks for all the replies, big fan of your various DS3 efforts over the years Nefarius!
